sdks/python/build.gradle

```diff
@@ -212,18 +209,57 @@ task installGcpTest(dependsOn: 'setupVirtualenv') {
     }
   }
 }
+installGcpTest.mustRunAfter sdist
+
+// Basic test options for ITs running on Jenkins.
+def basicTestOpts = [
+        "--nocapture",  // print stdout instantly
+        "--processes=8",  // run tests in parallel
+        "--process-timeout=4500", // timeout of whole command execution
+]
+
+def mapToArgString(argMap) {
+  def argList = []
+  argMap.each { k, v ->
+    if (v in List) {
+      v = "\"${v.join(' ')}\""
+    }
+    argList.add("--$k $v")
+  }
+  return argList.join(' ')
+}
 
 task directRunnerIT(dependsOn: 'installGcpTest') {
+  // Run IT tests with TestDirectRunner in batch.
   doLast {
+    def batchTestOpts = basicTestOpts + [
+        "--tests=apache_beam.examples.wordcount_it_test:WordCountIT.test_wordcount_it," +
+                "apache_beam.io.gcp.pubsub_integration_test:PubSubIntegrationTest," +
+                "apache_beam.io.gcp.big_query_query_to_table_it_test:BigQueryQueryToTableIT," +
+                "apache_beam.io.gcp.bigquery_io_read_it_test"
```

Comment:
Instead of embedding these tests in a string, can they be in an array and joined with ","?
How do I run only one of these tests? If you have them in an array, you could pass an override as  -P param?

Reply:
Array sounds good to me.

Another task `integrationTest` is available to run given set of pipelines with `-Ptests` or `-Pattr` param. Gradle task dependency is heavily used in Python PreCommit/PostCommit to group and run tests, it may not easy to abstract invocation part and pass -P param to setup and run different set of tests based on existing Gradle structure. Do you have a concrete example to help me understand?